inference-engine/thirdparty/clDNN/src/include/loop_inst.h

```diff
@@ -173,54 +237,73 @@ struct typed_program_node<loop> : public typed_program_node_base<loop> {
         }
     }
 
-    void process_single_int_input(const primitive_id& id) const {
+    void process_current_iteration() const {
+        const primitive_id& current_iteration_id = get_current_iteration_id();
+        if (current_iteration_id.empty()) {
+            return;
+        }
+
+        const topology_map& body_topology_map = body.get_primitives();
+        const layout body_input_layout(data_types::i64, format::bfyx, {1, 1, 1, 1});
+
+        // add current_iteration primitive if current_iteration primitive is not exist in body
+        if (body_topology_map.find(current_iteration_id) == body_topology_map.end()) {
+            body.add(std::make_shared<input_layout>(current_iteration_id, body_input_layout));
+        } else {
+            const auto& body_input_prim = body.at(current_iteration_id);
+            const auto input_layout_prim = std::dynamic_pointer_cast<input_layout>(body_input_prim);
+            if (!input_layout_prim) {
+                CLDNN_ERROR_MESSAGE(this->id(), "Not cldnn::input_layout");
```

Comment:
Could you use more descriptive exception message?

Reply:
Update message. thanks.